### ISSUE TITLE
Vault vault.read_secret fails if missing m2crypto

### ISSUE THREAD

**Issue by githubcdr** [saltstack/salt#47135]
### Description of Issue/Question

It seems that when M2Crypto is not installed the vault read_secret module fails with an RSA error.

### Setup

Latest Saltstack 2018.3.0 on master and minion.

### Steps to Reproduce Issue

```
zsh 8981 # salt ahdul vault.read_secret "secret/saltstack/global/pillar"
ahdul:
    ERROR: Exception occurred in runner vault.generate_token: Traceback (most recent call last):
      File "/usr/lib/python2.7/site-packages/salt/client/mixins.py", line 387, in _low
        data['return'] = self.functions[fun](*args, **kwargs)
      File "/usr/lib/python2.7/site-packages/salt/runners/vault.py", line 46, in generate_token
        _validate_signature(minion_id, signature, impersonated_by_master)
      File "/usr/lib/python2.7/site-packages/salt/runners/vault.py", line 127, in _validate_signature
        if not salt.crypt.verify_signature(public_key, minion_id, signature):
      File "/usr/lib/python2.7/site-packages/salt/crypt.py", line 245, in verify_signature
        return pubkey.verify(digest, signature)
      File "/usr/lib/python2.7/site-packages/M2Crypto/RSA.py", line 281, in verify
        return m2.rsa_verify(self.rsa, data, signature, digest_type)
    RSAError: bad signature
```

This is fixed after manually installing python2-m2crypto and restarting the minion.

On a fresh Windows minion this fails since m2crypto is not bundled with the installer?

### Versions Report

```
ahdul:
    Salt Version:
               Salt: 2018.3.0

    Dependency Versions:
               cffi: 0.8.6
           cherrypy: Not Installed
           dateutil: 1.5
          docker-py: 3.1.1
              gitdb: Not Installed
          gitpython: Not Installed
              ioflo: 1.3.7
             Jinja2: 2.7.2
            libgit2: Not Installed
            libnacl: 1.4.3
           M2Crypto: 0.21.1
               Mako: 0.9.1
       msgpack-pure: 0.1.3
     msgpack-python: 0.4.6
       mysql-python: 1.2.5
          pycparser: 2.10
           pycrypto: 2.6.1
       pycryptodome: Not Installed
             pygit2: Not Installed
             Python: 2.7.6 (default, Nov 23 2017, 15:49:48)
       python-gnupg: Not Installed
             PyYAML: 3.10
              PyZMQ: 15.2.0
               RAET: 0.6.3
              smmap: 0.8.2
            timelib: 0.2.4
            Tornado: 4.2.1
                ZMQ: 4.1.2

    System Versions:
               dist: Ubuntu 14.04 trusty
             locale: ascii
            machine: x86_64
            release: 3.13.0-143-generic
             system: Linux
            version: Ubuntu 14.04 trusty
```

**Comment by garethgreenaway** [comment]
@githubcdr Apologies for the delay in getting back to this. Do you have M2Crypto installed on the Salt master? I haven't been able to duplicate this is when the python m2crypto libraries are not installed at all.

**Comment by mchugh19** [comment]
The mix of m2crypto between salt-master and minion was fixed for us in #47379. I would expect this to be in the 2018 point release, but that is not out yet. You could test the patch, but it needs to be applied both on the salt-master as well as minions.
